### PR TITLE
Set undesired execution hosts to avoid in LSF

### PR DESCRIPTION
**Pull request by pcm32**
On our LSF setup we sometimes have nodes that are defective and we want to be up and running before the admin of the cluster can find the time to resolve them. This contribution enables the Galaxy admin to set a path to a file (that is read each time that a job is sent to the LSF cluster) where node names can be added, one per line, so that jobs avoid them on submission.

### CODE REVIEW

File: lib/galaxy/jobs/runners/util/cli/job/lsf.py [hunk 1]
```diff
@@ -125,5 +129,29 @@ def _get_job_state(self, state):
...
+            <param id="job_excluded_hosts">/path/to/file/with/hosts/to/exclude/one/per/line.txt</param>
+        </destination>
+
+        :param pathExcludedNodes:
+        :return: list with node names
+        """
+        if "excluded_hosts" in self.params:
+            path_excluded = self.params["excluded_hosts"]
+            if path.isfile(path_excluded):
+                return open(path_excluded).read().splitlines()
```
Comment (pcm32) [inline lib/galaxy/jobs/runners/util/cli/job/lsf.py]:
```suggestion
                with(open(path_excluded, 'r')) as f:
                    host_names = f.read().splitlines()
                return host_names
```

Comment (nsoranzo) [reply lib/galaxy/jobs/runners/util/cli/job/lsf.py]: You can even return from within the `with` statement:
```suggestion
                with open(path_excluded) as f:
                    return f.read().splitlines()
```

File: lib/galaxy/jobs/runners/util/cli/job/lsf.py [hunk 2]
```diff
@@ -125,5 +129,30 @@ def _get_job_state(self, state):
...
+
+        :param pathExcludedNodes:
+        :return: list with node names
+        """
+        if "excluded_hosts" in self.params:
+            path_excluded = self.params["excluded_hosts"]
+            if path.isfile(path_excluded):
+                with open(path_excluded) as f:
+                    return f.read().splitlines()
+        return list()
```
Comment (mvdbeek) [inline lib/galaxy/jobs/runners/util/cli/job/lsf.py]:
```suggestion
        return []
```

**Review by mvdbeek — APPROVED** [review]
Thanks @pcm32!

**Comment by pcm32** [comment]
Let me just try this as it is exactly written here in our LSF setup before merging, to make sure it is all good. Thanks for reviews and comments guys.

**Comment by mvdbeek** [comment]
If you're happy with this can you also resolve the merge conflict ?

**Comment by pcm32** [comment]
Resolved now, but I need to try this exact wording on our setup... bear with me.

**Comment by mvdbeek** [comment]
@pcm32 did you get a chance to test this ?

**Comment by pcm32** [comment]
@mvdbeek I have tested this locally as it is written here now, working well. Apologies for the delay.

**Comment by mvdbeek** [comment]
Thanks @pcm32!

**Pull request merged**